tracee-rules/signature.go

```diff
@@ -79,7 +79,10 @@ func findGoSigs(dir string) ([]types.Signature, error) {
 	return res, nil
 }
 
-func findRegoSigs(target string, partialEval bool, dir string) ([]types.Signature, error) {
+func findRegoSigs(target string, partialEval bool, dir string, aioEnabled bool) ([]types.Signature, error) {
```

Comment:
We need to add a way to filter Rego test files. In some of our input directories, test files are bundled with the rego polices in the same location. This pulls them in as well and as a result, fails during runtime. Example output:

```
sudo ../dist/tracee-ebpf -o format:gob --trace event=security_file_open,security_bprm_check,init_module,finit_module,magic_write,ptrace,process_vm_writev,execve,connect,setns,security_sb_mount,mem_prot_alert,commit_creds,bpf,mknod,mknodat,openat,security_inode_unlink,chmod,fchmod,chown,fchown,lchown,sched_process_exit,dup,dup2,dup3,close  | ./dist/tracee-rules --input-tracee file:stdin --input-tracee format:gob --rules-dir=inputDir/rego --rego-aio
2021/09/20 17:46:39 compiling modules: 11 errors occurred:
inputDir/rego/tests/change_namespace_test.rego:4: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/change_namespace_test.rego:10: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/command_history_deletion_test.rego:4: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/command_history_deletion_test.rego:18: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/command_history_deletion_test.rego:32: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/command_history_deletion_test.rego:46: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/container_absolute_path_discovery_test.rego:4: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/container_absolute_path_discovery_test.rego:23: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/container_absolute_path_discovery_test.rego:42: rego_unsafe_var_error: var tracee_match is unsafe
inputDir/rego/tests/cpu_fingerprint_test.rego:4: rego_unsafe_var_error: var tracee_match is unsafe
rego_compile_error: error limit reached


```

You can see how I handled it here to filter out by filenames: https://github.com/aquasecurity/tracee/pull/993/files#diff-c56a5239a886de2c9586c32e19abf0abcc3ae40b97437bc5fd3a9a78538281b5R112

Reply:
How is that working today without AIO? The same error? If yes we should fix it in dedicated PR

Reply:
Sure